src/store/modules/dashboard/type.ts

```diff
@@ -0,0 +1,36 @@
+
+export const VIEWERS_TYPE = Object.freeze({
+    PUBLIC: 'PUBLIC',
+    PRIVATE: 'PRIVATE',
+} as const);
+
+export type ViewersType = typeof VIEWERS_TYPE[keyof typeof VIEWERS_TYPE];
+
+export interface DashboardState {
+    domainItems: DomainDashboardItem[];
+    projectItems: ProjectDashboardItem[];
+}
+
+export interface DashboardItem {
+    name: string;
+    viewers: ViewersType;
+    version: number;
+    layouts: any[];
+    dashboard_options: any;
+    settings: any;
+    dashboard_options_schema: any;
+    labels: string[];
+    tags: any;
+    user_id: string;
+    domain_id: string;
+    created_at: string;
+    updated_at: string;
+}
+
+export interface DomainDashboardItem extends DashboardItem {
+    domain_dashboard_id: string;
+}
+export interface ProjectDashboardItem extends DashboardItem {
+    project_dashboard_id: string;
+}
+
```

Comment:
think: It's a bit vague where to manage the types of the dashboard. It would be nice to discuss and decide whether to put it in the dashboard service or store.

Reply:
I think so too... for now, it would be nice to reflect on it and think about it while working.